packages/core/src/internal/clients/cache/AbstractCacheClient.ts

```diff
@@ -751,6 +753,46 @@ export abstract class AbstractCacheClient implements ICacheClient {
     return await client.setSample(cacheName, setName, limit);
   }
 
+  /**
+   * Fetch and remove a random sample of elements from the set.
+   * Returns a different random sample for each call.
```

Comment:
I'm not clear what we mean by this comment? Is it that when we call set pop, we remove elements, hence can never get the same random sample twice? Or do we mean something else?